internal/primitives/core/hash/hash.go

```diff
@@ -0,0 +1,65 @@
+// Copyright 2024 ChainSafe Systems (ON)
+// SPDX-License-Identifier: LGPL-3.0-only
+
+package hash
+
+import (
+	"crypto/rand"
+	"encoding/binary"
+	"fmt"
+	"io"
+
+	"github.com/ChainSafe/gossamer/pkg/scale"
+)
+
+// H256 is a fixed-size uninterpreted hash type with 32 bytes (256 bits) size.
+type H256 string
```

Comment:
Why not use `[32]byte` since we already know it is a 32 byte array ?

Reply:
I need this to satisfy `constraints.Ordered` in future GRANDPA work as well as the client-db work.

Reply:
Ok! we have to create a golang proposal to improve the arrays comparison haha 😛